File: old_groovy/src/main/groovy/net/zomis/brainf/model/BrainfuckListener.java
package net.zomis.brainf.model;

import net.zomis.brainf.model.ast.tree.Syntax;

public interface BrainfuckListener {
    default void beforePerform(BrainfuckRunner runner, Syntax command) {}
    default void afterPerform(BrainfuckRunner runner, Syntax command) {}

    default void beforeWhile(BrainfuckRunner runner) {}
    default void afterWhile(BrainfuckRunner runner) {}
    default void beforeEndWhile(BrainfuckRunner runner) {}
    default void afterEndWhile(BrainfuckRunner runner) {}
}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainduck/runner/BrainfuckListener.kt
package net.zomis.brainduck.runner

import net.zomis.brainduck.analyze.BrainfuckRuntime
import net.zomis.brainduck.ast.Syntax

interface BrainfuckListener {

    fun before(syntax: Syntax, runtime: BrainfuckRuntime)
    fun after(syntax: Syntax, runtime: BrainfuckRuntime)

}